elip-deterministic-blinding-key.mediawiki

```diff
@@ -0,0 +1,133 @@
+
+Potential adopters of this standard should consider whether this behavior is acceptable.
+
+===Specification===
+The following descriptor:
+
+<pre>
+ct(elipXXXX,<DESCRIPTOR>)
+</pre>
+
+Is equivalent to:
+
+<pre>
+ct(<KEY>,<DESCRIPTOR>)
+</pre>
+
+Where <code>KEY</code> is a ELIP-150 view blinding key derived as follows:
+* If the top-level script expression is <code>combo</code>, fail.
+* Future extensions that support multiple descriptors other than multi-path are not supported.
+* If <code>DESCRIPTOR</code> is multi-path, expand to every single descriptor.
+* For each single descriptor:
+** If it has wildcards, get the definite descriptor at index 2<sup>31</sup>-1<ref>
+We want to use a <code>scriptPubKey</code> that will not be used in a transaction.
+Otherwise it can be used directly by anyone to derive the descriptor blinding key.
+Wallets use <code>scriptPubKey</code>s starting from index 0,
+thus we choose the last non-hardened index,
+which in practice will never be used in a transaction.
+</ref>
+** Derive the <code>scriptPubKey</code> from the definite descriptor<ref>
+Here we use the <code>scriptPubKey</code> because descriptors have fields that might change without affecting the spending conditions.
+For instance changing the fingerprint of one xpub yields the same <code>scriptPubKey</code>s.
+Using the <code>scriptPubKey</code> allows us to treat such descriptors as equivalent.
+</ref>
+* Concatenate all <code>scriptPubKey</code>s obtained before, separating them with <code>OP_INVALIDOPCODE</code>.
+* Let ''H<sub>tag</sub>'' be a [https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki#Design BIP-340 tagged hash] with tag <code>Deterministic-View-Key/1.0</code><ref>
+A tagged hash is used to make sure hashes used in this context can't be reinterpreted in another one.
+</ref>
+* The descriptor blinding key is the tagged hash of the concatenated <code>scriptPubKey</code>s, ''<code>KEY</code> = H<sub>tag</sub>(<code>scriptPubKey</code>, <code>OP_INVALIDOPCODE</code>, ...) mod n'', converted to WIF format.
+
+==Test Vectors==
+
+The following ordinary descriptor yield to the given descriptor blinding key:
+
+* Test vector: 1
+** Ordinary descriptor: <code>elwpkh(xpub661MyMwAqRbcFkPHucMnrGNzDwb6teAX1RbKQmqtEF8kK3Z7LZ59qafCjB9eCRLiTVG3uxBxgKvRgbubRhqSKXnGGb1aoaqLrpMBDrVxga8/<0;1>/*)#lg37z46y</code>
+** Confidential descriptor (equivalent version): <code>ct(elipxxxx,elwpkh(xpub661MyMwAqRbcFkPHucMnrGNzDwb6teAX1RbKQmqtEF8kK3Z7LZ59qafCjB9eCRLiTVG3uxBxgKvRgbubRhqSKXnGGb1aoaqLrpMBDrVxga8/<0;1>/*))#cawe8rjm</code>
+** Confidential descriptor: <code>ct(L3TsJMSZ1gfJyLgyT16Pb8c7Tj1oxosPvZGoHpryH8frpXf6Sphb,elwpkh(xpub661MyMwAqRbcFkPHucMnrGNzDwb6teAX1RbKQmqtEF8kK3Z7LZ59qafCjB9eCRLiTVG3uxBxgKvRgbubRhqSKXnGGb1aoaqLrpMBDrVxga8/<0;1>/*))#0aphatv9</code>
+** Descriptor blinding key: <code>L3TsJMSZ1gfJyLgyT16Pb8c7Tj1oxosPvZGoHpryH8frpXf6Sphb</code>
```

Comment:
nit: (IMO) `Descriptor blinding key` -> `Derived descriptor blinding key`,  `Confidential descriptor` -> `Derived confidential descriptor`, then swap the order (derived  descriptor blinding key should come first).

Reply:
Changed